charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -243,11 +243,7 @@ spec:
           name: consul-data
           readOnly: false
         {{- if .Values.global.tls.enabled }}
```

Comment:
I think this needs similar logic for not mounting the ca cert if external servers with `useSystemRoots` are enabled (like on line 180)

Reply:
👍 without the mount skip, this seems to cause problems when `useSystemRoots` are enabled due to the cert secrets referenced here not existing. I'm assuming it means we don't bootstrap the CA/self-signed cert when that flag is enabled since we know we aren't going to use it?

Reply:
We only bootstrap the cert if servers are enabled (https://github.com/hashicorp/consul-k8s/blob/ef421f968770e96ee2cbbb471423169b579aa1de/charts/consul/templates/tls-init-job.yaml#L1). In this case, servers are external so we wouldn't bootstrap it either way. We'd either expect to be provided via a secret or ignore it if `useSystemRoots` is `true`